services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainMetrics.java

```diff
@@ -173,15 +194,35 @@ private void updateMetrics(final BeaconState state) {
   }
 
   private int getLiveValidators(final SSZList<PendingAttestation> attestations) {
-    final Map<UInt64, Map<UInt64, Bitlist>> aggregationBitsBySlotAndCommittee = new HashMap<>();
-    attestations.forEach(
+    return getNumberOfValidators(attestations, (__) -> true);
```

Comment:
I think we should avoid iterating through all the attestations twice. We just need to iterate through once, counting both total validators and correct validators.  Likely that will be easiest if you just an old fashioned for loop rather than streams.